src/lib/codegenerator/generator.js

```diff
@@ -287,7 +287,9 @@ const generateForLoopCode = function (templateObject, parent) {
     const created${forStartCounter} = []
     const forloop${forStartCounter} = (collection = [], elms, created) => {
       const rawCollection = getRaw(collection)
-      const keys = new Set(rawCollection.map((${item}) => '' +  ${interpolate(key, '')}))
+      const keys = new Set(rawCollection.map((${item}, index) => ${
+    key ? interpolate(key, '') : 'index'
```

Comment:
@suresh-gangumalla should 'index' indeed be a string? or did you mean to use the variable `index` ?

Reply:
To use `index` as variable only

Reply:
As per current `dev` code, when we avoid/forgot key attribute in forloop then `__index` is treated as key so the 'index' here will reference to index variable of map in generated code. 

<img width="298" alt="forloop-default-key-2" src="https://github.com/user-attachments/assets/bee481c6-3283-45b6-aa22-41c6af76f210">